oidc.go

```diff
@@ -307,3 +313,16 @@ func (j *jsonTime) UnmarshalJSON(b []byte) error {
 	*j = jsonTime(time.Unix(unix, 0))
 	return nil
 }
+
+func unmarshalResp(r *http.Response, body []byte, v interface{}) error {
+	err := json.Unmarshal(body, &v)
+	if err == nil {
+		return nil
+	}
+	ct := r.Header.Get("Content-Type")
+	mediaType, _, _ := mime.ParseMediaType(ct)
+	if mediaType != "application/json" {
+		return fmt.Errorf("expected Content-Type = application/json, got %q", ct)
+	}
+	return fmt.Errorf("expected Content-Type = application/json, but could not parse media type out of %q: %v", ct, err)
```

Comment:
This is the wrong error. If you failed to parse the media type, `mediaType` would be "". This can only be hit when you actually correctly parse the media type and it's "application/json"